daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c

```diff
@@ -379,14 +385,19 @@ static int ipapwd_pre_add(Slapi_PBlock *pb)
     pwdop->pwdata.timeNow = time(NULL);
     pwdop->pwdata.target = e;
 
-    ret = ipapwd_CheckPolicy(&pwdop->pwdata);
-    /* For accounts created by cn=Directory Manager or a passsync
-     * managers, ignore result of a policy check */
-    if ((pwdop->pwdata.changetype != IPA_CHANGETYPE_DSMGR) &&
-        (ret != 0) ) {
-        errMesg = ipapwd_error2string(ret);
-        rc = LDAP_CONSTRAINT_VIOLATION;
-        goto done;
+    /* If the entry represents an app password, no password policy checks
+     * are performed
+     */
+    if (strstr(pwdop->pwdata.dn, 'cn=apps,cn=accounts,dc=') == NULL) {
```

Comment:
A string compare of a DN is probably not the best way to do this.

Reply:
I refactored it to use slapi_ldap_explode_dn() function. It is now much more complex both comutationally and w.r.t. code, but it should hopefully be more robust. I now also check for OOM.